tests/datasets/test_eurosat.py

```diff
@@ -60,19 +60,58 @@ def test_add(self, dataset: EuroSAT) -> None:
         assert len(ds) == 4
 
     def test_already_downloaded(self, dataset: EuroSAT, tmp_path: Path) -> None:
-        type(dataset)(tmp_path)
+        type(dataset)(tmp_path, split=dataset.split)
 
     def test_already_downloaded_not_extracted(
         self, dataset: EuroSAT, tmp_path: Path
     ) -> None:
         shutil.rmtree(dataset.root)
         shutil.copy(dataset.url + dataset.filename, tmp_path)
-        type(dataset)(tmp_path)
+        type(dataset)(tmp_path, split=dataset.split)
 
     def test_not_downloaded(self, tmp_path: Path) -> None:
         with pytest.raises(DatasetNotFoundError, match='Dataset not found'):
             EuroSAT(tmp_path)
 
+    def test_missing_images_and_zip_no_download(self, tmp_path: Path) -> None:
+        """Ensure DatasetNotFoundError is raised if images and zip are missing and download=False."""
+        split_file = tmp_path / 'eurosat-train.txt'
+        split_file.write_text('dummy.tif\n')
+        with pytest.raises(DatasetNotFoundError):
+            EuroSAT(root=tmp_path, split='train', download=False)
+
+    def test_image_folder_present_split_file_missing(
+        self, tmp_path: Path, monkeypatch: MonkeyPatch
+    ) -> None:
+        """Test that split file is downloaded if missing but image folder is present."""
+        image_dir = (
+            tmp_path
+            / 'ds'
+            / 'images'
+            / 'remote_sensing'
+            / 'otherDatasets'
+            / 'sentinel_2'
+            / 'tif'
+        )
+        class_dir = image_dir / 'AnnualCrop'
+        class_dir.mkdir(parents=True, exist_ok=True)
+        (class_dir / 'dummy.tif').touch()
+
+        split_file = tmp_path / 'eurosat-train.txt'
+        if split_file.exists():
+            split_file.unlink()
+
+        called = {}
+
+        def fake_download_url(url: str, root: str, md5: str) -> None:
+            called['url'] = url
+            split_file.write_text('dummy.tif\n')
+
+        monkeypatch.setattr('torchgeo.datasets.eurosat.download_url', fake_download_url)
```

Comment:
I don't love this, can't we monkeypatch the URL to the fake data folder and download (copy) the real file? We already monkeypatch download_url to copy a local file.

Reply:
Not sure I follow, the monkeypatch download_url would copy everything in the tests/data/eurosat folder (archive, image folder and split files). We're looking to have the directory have just the image folder (ds) before downloading.

Reply:
I'm suggesting to remove this line, monkeypatch `url`, and instead use `EuroSAT(download=True)`. Then the dataset will be doing the downloading (copying) for us.